selfdrive/car/toyota/carstate.py

```diff
@@ -4,16 +4,21 @@
 from selfdrive.car.interfaces import CarStateBase
 from opendbc.can.parser import CANParser
 from selfdrive.config import Conversions as CV
-from selfdrive.car.toyota.values import CAR, DBC, STEER_THRESHOLD, TSS2_CAR, NO_DSU_CAR, NO_STOP_TIMER_CAR
+from selfdrive.car.toyota.values import CAR, DBC, STEER_THRESHOLD, NO_DSU_CAR, TSS2_CAR, NO_STOP_TIMER_CAR
 
 
 class CarState(CarStateBase):
   def __init__(self, CP):
     super().__init__(CP)
     can_define = CANDefine(DBC[CP.carFingerprint]['pt'])
     self.shifter_values = can_define.dv["GEAR_PACKET"]['GEAR']
+
     self.angle_offset = 0.
-    self.init_angle_offset = False
+
+    # Only learn angle offset in non TSS2 cars without DSU
+    learn_angle_offset = CP.carFingerprint in NO_DSU_CAR and CP.carFingerprint not in TSS2_CAR
+    self.init_angle_offset = not learn_angle_offset
```

Comment:
minor: `not` in this line and `not` in line 57. You can probably pick a different var name and avoid the double `not`